### PR TITLE
Routing POC

### PR DESCRIPTION
**Pull request by jaredperreault-okta**
## Proposed solution: 
* __[BREAKING CHANGE]__ Remove `SecureRoute` from SDK (and the `react-router`, `react-router-dom` peer dependencies).
  * _This change is not included in PR_
* Provide a series of samples, illustrating various implementations of routing, using different popular react routing libraries (currently located [here](https://github.com/okta/okta-react/tree/jp-OKTA-447325/test/apps/react-routing))

### CODE REVIEW

File: test/apps/react-routing/reach-router/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@okta/test.app.reach-router",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@okta/okta-auth-js": "5",
```
[the remark this answers is not in the document]
Comment (jaredperreault-okta) [reply test/apps/react-routing/reach-router/package.json:6]: good catch

File: test/apps/react-routing/reach-router/src/components/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+
+const Protected = () => {
```
[the remark this answers is not in the document]
Comment (jaredperreault-okta) [reply test/apps/react-routing/reach-router/src/components/Login.tsx:16]: This component isn't actually used currently, but I'll fix it

File: test/apps/react-routing/react-router-dom-v5/src/components/SecureRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+import { toRelativeUrl } from '@okta/okta-auth-js';
+import { Route, RouteProps, useRouteMatch } from 'react-router-dom';
+
+export const SecureRoute: React.FC<{
```
Comment (oleksandrpravosudko-okta) [inline test/apps/react-routing/react-router-dom-v5/src/components/SecureRoute.tsx:18]: would it be feasible to ship both `SecureRoute` and `SecureRoutes` under respective router subdirectories as a part of SDK? (we could probably mark `reach-router` and `react-router-dom` as [optional](https://yarnpkg.com/configuration/manifest#peerDependenciesMeta) peer dependencies)

Comment (jaredperreault-okta) [reply test/apps/react-routing/react-router-dom-v5/src/components/SecureRoute.tsx:18]: Our plan is to remove `SecureRoute` from the sdk altogether and then use these test apps as samples/examples of how one would implement routing with the various react routing libraries

Comment (tehhowch) [reply test/apps/react-routing/react-router-dom-v5/src/components/SecureRoute.tsx:19]: any movement here?

Comment (jaredperreault-okta) [reply test/apps/react-routing/react-router-dom-v5/src/components/SecureRoute.tsx:19]: @tehhowch yes, https://github.com/okta/okta-react/tree/master/samples/routing

**Pull request closed without merging**